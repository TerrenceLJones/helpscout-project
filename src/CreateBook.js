import React, { Component } from 'react';

export default class CreateBook extends Component {
  state = {
    title: '',
    category: '',
    image: ''
  }

  cancelAdd = () => {
    this.props.history.goBack();
  };

  handleTitleChange = ({ target: { value } }) => {
    this.setState({ title: value })
  }

  handleCategoryChange = ({ target: { value } }) => {
    this.setState({ category: value })
  }

  handleImageChange = ({ target: { files } }) => {
    let reader = new FileReader();
    let file = files[0];

    reader.onloadend = () => {
      this.setState({
        image: file,
        imagePreviewUrl: reader.result
      });
    }

    reader.readAsDataURL(file)
  }

  handleSubmit = (e) => {
    e.preventDefault();
    console.log(this.state);
  }

  getImagePreview = () => {
    let { imagePreviewUrl } = this.state;

    if (!imagePreviewUrl) {
      return <div>Please select an Image for Preview</div>;
    }

    return <img src={ imagePreviewUrl } alt="Your book's cover image."/>;
  }

  render() {
    return (
      <div>
        <h1>Create a New Book</h1>

        <button onClick={ this.cancelAdd }>Cancel</button>

        <form onSubmit={ this.handleSubmit }>

          <label htmlFor="title">Title</label>
          <input
            type="text"
            id="title"
            value={ this.state.title }
            onChange={ this.handleTitleChange }
          />

          <label htmlFor="category">Category</label>
          <input type="text"
            id="category"
            value={ this.state.category }
            onChange={ this.handleCategoryChange }
          />

          { this.getImagePreview() }

          <input
            type="file"
            id="image"
            accept="image/*"
            onChange={ this.handleImageChange }
          />

          <button type="submit">Create Book</button>
        </form>
      </div>
    );
  }
};
