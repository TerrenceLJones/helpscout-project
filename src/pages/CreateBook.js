import React from 'react';

import CreateBookFrom from 'components/BookForm';

const CreateBook = () => <CreateBookFrom />;

export default CreateBook;
