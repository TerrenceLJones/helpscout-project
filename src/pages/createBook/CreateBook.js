import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import CreateBookFrom from 'components/bookForm';

import { booksActions } from 'state/modules/books';

const CreateBook = ({ createBook, history }) => {

  const handleSubmit = (bookValues) => {
    createBook(bookValues)
      .then(({ payload: { data: newBook } }) => {
        history.replace(`/books/${newBook.id}`);
      });
  }

  const getHeader = () => {
    return (
      <header>
        <h1>Create a New Book</h1>
        <Link to={ `/books` }>Cancel</Link>
      </header>
    );
  }

  return (
    <div>
      { getHeader() }

      <CreateBookFrom handleSubmit={ handleSubmit } />
    </div>
  );
}

CreateBook.propTypes = {
  createBook: PropTypes.func,
  history: PropTypes.object
};

const mapDispatchToProps = dispatch => ({
  createBook: bookData => dispatch(booksActions.createBook(bookData)),
});

export {
  CreateBook
}

export default connect(null, mapDispatchToProps)(CreateBook);
