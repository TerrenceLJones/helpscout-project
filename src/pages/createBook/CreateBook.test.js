import React from 'react';
import { shallow } from 'enzyme';

import { CreateBook } from './CreateBook';


it('renders without crashing', () => {
  expect(shallow(<CreateBook
    createBook={ jest.fn() }
    history={ { replace: jest.fn() } }
  />)).toMatchSnapshot();
});

it('navigates to book details after submission', () => {});
