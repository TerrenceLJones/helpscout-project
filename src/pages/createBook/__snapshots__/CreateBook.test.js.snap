// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CreateBook
    createBook={[MockFunction]}
    history={
      Object {
        "replace": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header>
          <h1>
            Create a New Book
          </h1>
          <Link
            replace={false}
            to="/books"
          >
            Cancel
          </Link>
        </header>,
        <BookForm
          handleSubmit={[Function]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1>
              Create a New Book
            </h1>,
            <Link
              replace={false}
              to="/books"
            >
              Cancel
            </Link>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Create a New Book",
            },
            "ref": null,
            "rendered": "Create a New Book",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Cancel",
              "replace": false,
              "to": "/books",
            },
            "ref": null,
            "rendered": "Cancel",
            "type": [Function],
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header>
            <h1>
              Create a New Book
            </h1>
            <Link
              replace={false}
              to="/books"
            >
              Cancel
            </Link>
          </header>,
          <BookForm
            handleSubmit={[Function]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1>
                Create a New Book
              </h1>,
              <Link
                replace={false}
                to="/books"
              >
                Cancel
              </Link>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Create a New Book",
              },
              "ref": null,
              "rendered": "Create a New Book",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Cancel",
                "replace": false,
                "to": "/books",
              },
              "ref": null,
              "rendered": "Cancel",
              "type": [Function],
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
